File: src/main/java/inno/jago/ast/ImportNode.java
package inno.jago.ast;

import inno.jago.ast.entity.Entity;

/*
    importDecl          :   IMPORT ( importSpec | L_PAREN ( importSpec END )* ( importSpec ( END )? )? R_PAREN );
    importSpec          :   ( DOT | packageName )? importPath;
    importPath          :   STRING_LIT;
 */
public class ImportNode extends Entity {

    private String path;
    private String alias;
    private boolean importAll;

    // todo: add constructor based on optionality
}

File: src/main/kotlin/inno/jago/ast/global/ImportNode.kt
package inno.jago.ast.global

import inno.jago.ast.ASTNode
import inno.jago.lexer.Pos

class ImportNode(
    pos: Pos,
    val path: String,
    val alias: String?,
    val importAll: Boolean
) : ASTNode(pos = pos) {
    val name = path.split("/").last()
}
